src/cascadia/TerminalSettingsModel/ActionMapSerialization.cpp

```diff
@@ -78,7 +78,9 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             // Now check if this is a command block
             if (jsonBlock.isMember(JsonKey(CommandsKey)) || jsonBlock.isMember(JsonKey(ActionKey)))
             {
-                AddAction(*Command::FromJson(jsonBlock, warnings, origin), keys);
+                auto command = Command::FromJson(jsonBlock, warnings, origin);
+                command->LogSettingChanges(_changeLog);

```

Comment:
Do we need an `origin != InBox` check here? Otherwise we're calling `LogsettingChanges` on all the `InBox` actions being layered here

Reply:
We don't _need_ to add the `InBox` check because we're looping over the json block, so we're just reporting the ones in settings.json.

That said, this whole PR does have the problem that we're still adding a bunch of these changes to `_changeLog` in the in box layer. They just don't get reported because of inheritance (we're calling `LogSettingChanges` on the `CascadiaSettings` from settings.json, not defaults.json). It'd be nice if we didn't do the extra work/storage, but I'm not sure how to accomplish that. 🤔